src/Database/Transactions/GenericTransactionManager.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database\Transactions;
+
+use PDO;
+use Tempest\Database\Exceptions\CannotBeginTransaction;
+use Tempest\Database\Exceptions\CannotCommitTransaction;
+use Tempest\Database\Exceptions\CannotRollbackTransaction;
+
+final class GenericTransactionManager implements TransactionManager
+{
+    public function __construct(private PDO $pdo)
+    {
+    }
+
+    public function begin(): void
+    {
+        $transactionBegun = $this->pdo->beginTransaction();
+
+        if (! $transactionBegun) {
+            throw new CannotBeginTransaction();
```

Comment:
Technically, this exception should be called `CouldNotBeginTransaction`, since we've tried to start it, and it didn't work. The current name implies that we didn't even try to start it, because of some conditions unknown.